pkg/test/echo/client/client.go

```diff
@@ -33,14 +35,23 @@ type Instance struct {
 }
 
 // New creates a new echo client.Instance that is connected to the given server address.
-func New(address string) (*Instance, error) {
+func New(address string, tlsSettings *common.TLSSettings) (*Instance, error) {
 	// Connect to the GRPC (command) endpoint of 'this' app.
 	ctx, cancel := context.WithTimeout(context.Background(), common.ConnectionTimeout)
 	defer cancel()
-	conn, err := grpc.DialContext(ctx,
-		address,
-		grpc.WithInsecure(),
-		grpc.WithBlock())
+	dialOptions := []grpc.DialOption{grpc.WithBlock()}
+	if tlsSettings == nil {
+		dialOptions = append(dialOptions, grpc.WithInsecure())
+	} else {
+		cert, err := tls.X509KeyPair([]byte(tlsSettings.ClientCert), []byte(tlsSettings.Key))
+		if err != nil {
+			return nil, err
+		}
+
+		cfg := credentials.NewTLS(&tls.Config{InsecureSkipVerify: true, Certificates: []tls.Certificate{cert}})
```

Comment:
why skip verify when mtls?

Reply:
I could not get this to work properly through the port forward since the cert doesn't match the port forward address. If we can get it working that would be nice though

Reply:
Interesting, can you provide more details ? The cert should be spiffee - and not depend on address.

Reply:
Ill send a fix out in another PR

Reply:
https://github.com/istio/istio/pull/22177